src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -1751,18 +1738,7 @@ void MergeTreeData::loadDataParts(bool skip_sanity_checks, std::optional<std::un
                 continue;
             }
 
-            bool is_disk_defined = defined_disk_names.contains(disk_name);
-
-            if (!is_disk_defined && disk->exists(relative_data_path))
-            {
-                /// There still a chance that underlying disk is defined in storage policy
-                const auto & delegate = disk->getDelegateDiskIfExists();
-                is_disk_defined = delegate && !delegate->isBroken() && !delegate->isCustomDisk()
-                               && delegate->getPath() == disk->getPath()
-                               && defined_disk_names.contains(delegate->getName());
```

Comment:
This should be restored, otherwise, encrypted disks are broken:
https://pastila.nl/?00404443/3a71358d4433a0fccb0eb72190911111#jTQnOEO9WCKEw3HITu8ukw==